tracks/julia/exercises/word-count/mentoring.md

```diff
@@ -121,4 +121,94 @@ function wordcount(sentence)
 end
 ```
 
+** Without Regex ** 
+
+This solution doesn't use a regex at all. It uses the default split behavior (which checks for `isspace` characters) and then strips punctuation characters left over on the "outside" of words. Filtering hoas to come after stripping in case the strip operation itself left an empty string.
+```julia
+function wordcount(sentence)
+    words = strip.(ispunct, split(lowercase(sentence)))
+    filter!(!isempty, words)
+    d = Dict{eltype(words), Int}()
+    for w in words
+        d[w] = get(d, w, 0) + 1
+    end
+    d
+end
+```
+It's possible to create the dictionary more efficiently with the `countmap` function from `StatsBase` (which only hashes the key once per check). With countmap, you can code-golf the above into:
+```julia
+wordcount_withstatsbase(sentence) = countmap(filter!(!isempty, strip.(ispunct, split(lowercase(sentence)))))
+```
+
+
+** Totally manual ** 
+
+This method iterates through the string one character at a time and switches behavior at word boundaries (when `iswordchar` changes from true to false or false to true). The dictionary has `SubString` keys in this version, which are like the string-version of Array `views`.
+```julia
+function wordcount(sent)
+
+    sent = lowercase(sent)
+    d = Dict{SubString, Int}()
```

Comment:
Should this be SubString{String} or is that the default?

Reply:
```suggestion
    d = Dict{SubString{typeof(sent)}, Int}()
```

```julia
julia> view("hello", 1:2) |> typeof
SubString{String}
```

Reply:
Maybe `sent` should be `input` for consistency, too.

Reply:
The `view` version only works on v1.6 I think. At least it doesn't for me on 1.2 and 1.5

I agree `input` would be preferable. I'll change it.

Reply:
Also, it should be `SubString{String}`! Thanks for that correction, I forgot to respond to it!